src/include/ipc/topology.h

```diff
@@ -126,6 +128,13 @@ struct sof_ipc_comp_mixer {
 	struct sof_ipc_comp_config config;
 } __attribute__((packed));
 
+/* generic smart amplifier component */
+struct sof_ipc_comp_smart_amp {
+	struct sof_ipc_comp comp;
+	struct sof_ipc_comp_config config;
+	uint32_t feedback_buf_id;	/**< buffer feedback index */
```

Comment:
what does this index represent? The component ID of the buffer? the topology ID?
There are so many indices throughout topology, kernel and firmware, you need to be more explicit on what this is tied to.

Reply:
Sorry @plbossart , I've just updated PR and commit messages.

I've talked with @xiulipan and we decided to use ```sof_ipc_comp_process``` struct
(like e.g. in KPB; it does not have ```feedback_buf_id``` variable) instead of ```sof_ipc_comp_smart_amp```.